site/content/contribute/server/_index.md

```diff
@@ -20,10 +20,31 @@ The server consists of several different Go packages:
 
 * `api4` - Version 4 of the RESTful JSON Web Service
 * `app` - Logic layer for getting, modifying, and interacting with models
+* `services` - Set of packages providing functionality to the rest of the application, especially to the `app` package.
 * `cmd` - Command line interface
 * `einterfaces` - Interfaces for Enterprise Edition features
 * `jobs` - Job server and scheduling
 * `model` - Definitions and helper functions for data models
 * `store` - Storage layer for interacting with caches and databases
 * `utils` - Utility functions for various tasks
 * `web` - Serves static pages
+
+## New packages rules
+
+We are trying to follow certain rules about when is needed to create a new
+package or add functionality to the existing ones (Some of them can not fit
+with the current implementation because it is an ongoing effort).
+
+* Any integration with a third party tool should be implemented as an
+  independent service (Examples: Bleve, Cache, File backends).
+* Any well isolated chunk of code that has a very small public API and a lot of
+  code behind, should be separated into a service (Examples: Import/Export,
+  Tracing, Telemetry)
+* If you are adding logic that interact with differnt services, put the
+  information together, and give a result, probably you are adding logic to the
+  `app` package.
+* If you are validating inputs from the user or permissions, you should do it
+  in the external layers (`api4`, `web`, `app/slashcommands`).
+* If you are initializing something (for example a new service) that is going
+  to be there for the whole time the server is running, you should be adding it
+  to the `app.Server` data structure, probably in the `app/server.go` file.
```

Comment:
We can add a sentence like: "You should be aware of the life cycle of any entity you instantiate in the code."

Reply:
That is an interesting advice and for sure should be true, I'm not 100% sure if this is the place to reiterate that.

Reply:
Yeah, probably not. Just wanted to remind it, again :)